include/ccf/crypto/pem.h

```diff
@@ -16,14 +16,25 @@ namespace crypto
   // Convenience class ensuring null termination of PEM-encoded certificates
   class Pem
   {
+  private:
     std::string s;
 
+    void check_pem_format()
+    {
+      if (!s.starts_with("-----"))
+      {
+        throw std::runtime_error(
+          fmt::format("PEM constructed with non-PEM data: {}", s));
```

Comment:
That doesn't seem to be a very thorough check that the data is in fact PEM-encoded?

Reply:
It's not, but it's good enough to catch all of our current failure cases (empty strings, DERs). I looked at OpenSSL's `PEM_read_bio`, which would do more thorough validation, but would (afaict) still pass things that might fail when we actually try to use them (ie - as certs, or keys) later.

Reply:
Let's gate that on NDEBUG then?

Reply:
I think we want this in release/NDEBUG builds too as it catches simple errors earlier. Without that, users may get cryptic OpenSSL messages that they can't make sense of (e.g. "ASN.1 content too short", without information about where that ASN.1 came from and why it's too short).

Reply:
But that's at the expense of running this string comparison on every single Pem object construction, which seems really arbitrary and expensive. If we want to validate input, like when parsing JSON, we should do it all the way, not like this.